### PR TITLE
Add ignore_empty for optional uri field validation

### PR DESCRIPTION
**Pull request by emcfarlane**
Adds to all uri fields that can be optional set. Using with options still adds ignore_empty to allow unsetting the value.

Otherwise empty string "" will fail validation with: 
```
invalid_argument: validation error:
 - values[0].url: value must be a valid URI [string.uri]
```

### CODE REVIEW

File: buf/registry/owner/v1beta1/user_service.proto [hunk 1]
```diff
@@ -107,7 +107,8 @@ message CreateUsersRequest {
     // The configurable URL that represents the homepage for a User.
     string url = 4 [
       (buf.validate.field).string.uri = true,
-      (buf.validate.field).string.max_len = 255
+      (buf.validate.field).string.max_len = 255,
+      (buf.validate.field).ignore_empty = true
```
Comment (mfridman) [inline buf/registry/owner/v1beta1/user_service.proto:111]: Curious, why does this have to be empty? There is no `min_len` set on the field?

Comment (jchadwick-buf) [reply buf/registry/owner/v1beta1/user_service.proto:111]: Because otherwise the uri validator fails: it does not accept an empty value as a valid URI.

**Review by jchadwick-buf — APPROVED** [review]
Tested running protovalidate on the user create RPC and user update RPC. This seems to do what is intended.

File: buf/registry/owner/v1beta1/user_service.proto (hunk 1 above)
Comment (mfridman) [reply buf/registry/owner/v1beta1/user_service.proto:111]: Got it! That makes sense.

Comment (emcfarlane) [reply buf/registry/owner/v1beta1/user_service.proto:111]: An empty string isn't a valid URI so fails the string on string uri: `values[0].url: value must be a valid URI [string.uri]`.

File: buf/registry/module/v1beta1/module_service.proto [hunk 2]
```diff
@@ -146,7 +147,8 @@ message UpdateModulesRequest {
     // uThe configurable URL in the description of the module.
```
Comment (mfridman) [inline buf/registry/module/v1beta1/module_service.proto:147]: Typo (unrelated to PR).

Comment (emcfarlane) [reply buf/registry/module/v1beta1/module_service.proto]: Nice catch!

**Review by bufdev — APPROVED** [review]

**Pull request merged**